systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/PodUtils.java

```diff
@@ -187,17 +187,32 @@ public static void waitUntilMessageIsInLogs(String podName, String containerName
 
     public static void waitUntilPodContainersCount(String podNamePrefix, int numberOfContainers) {
         LOGGER.info("Wait until Pod {} will have {} containers", podNamePrefix, numberOfContainers);
-        TestUtils.waitFor("Pod" + podNamePrefix + " will have " + numberOfContainers + " containers",
+        TestUtils.waitFor("Pod " + podNamePrefix + " will have " + numberOfContainers + " containers",
             Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_STATUS_TIMEOUT,
             () -> kubeClient().listPodsByPrefixInName(podNamePrefix).get(0).getSpec().getContainers().size() == numberOfContainers);
         LOGGER.info("Pod {} has {} containers", podNamePrefix, numberOfContainers);
     }
 
-    public static void waitUntilPodReplicasCount(String podNamePrefix, int exceptedPods) {
-        LOGGER.info("Wait until Pod {} will have {} replicas", podNamePrefix, exceptedPods);
+    public static void waitUntilPodStabilityReplicasCount(String podNamePrefix, int exceptedPods) {
+        LOGGER.info("Wait until Pod {} will have stable {} replicas", podNamePrefix, exceptedPods);
+        int[] stableCounter = {0};
```

Comment:
I guess you are using this one element array because otherwise, you cannot increment the variable in the lambda. Maybe you can use an AtomicInteger? Or there is a specific reason for using the array?

Reply:
Using AtomicInteger is overloading for no such benefit. Usually, I do not use AtomicInteger, unless I have a really good reason to use and the reason for using AtomicInteger might be only allowing concurrent accesses. Using the static array to have reference inside lambda with one element has many benefits over AtomicInteger and it's low cost.